tests/syslog/test_syslog.py

```diff
@@ -1,62 +1,66 @@
+    pytest.mark.topology("any")
 ]
 
+
 @pytest.fixture(scope="module")
-def config_syslog_srv(ptfhost):
-    logger.info( "Configuring the syslog server")
+def config_syslog_srv(ptfhost, duthost):
+    logger.info("Configuring the syslog server")
 
-    # add the imudp configuration if not present
-    ptfhost.shell('sed -ni \'/module/!p;$a module(load="imudp")\' /etc/rsyslog.conf')
-    ptfhost.shell('sed -i \'/input(type/!p;$a input(type="imudp" port="514")\' /etc/rsyslog.conf')
+    # Add the imudp configuration if not present
+    ptfhost.shell("sed -ni '/module/!p;$a module(load=\"imudp\")' /etc/rsyslog.conf")
+    ptfhost.shell("sed -i '/input(type/!p;$a input(type=\"imudp\" port=\"514\")' /etc/rsyslog.conf")
 
-    #Remove local /var/log/syslog   
+    # Remove local /var/log/syslog
     ptfhost.shell("rm -rf /var/log/syslog")
 
-    #restart Syslog Daemon 
+    # Restart Syslog Daemon
     ptfhost.shell("service rsyslog restart")
 
-    #Wait a little bit for service to start
-    time.sleep(30)
+    # Wait a little bit for service to start
+    def _is_syslog_running():
+        result = duthost.shell("service rsyslog status | grep \"active (running)\"")["stdout_lines"]
+        return len(result) > 0
+
+    wait_until(30, 3, _is_syslog_running)
 
     yield
 
+
 @pytest.fixture(scope="module")
 def config_dut(testbed, duthost):
-    logger.info( "Configuring the DUT")
-    local_syslog_srv_ip = testbed['ptf_ip']
-    logger.info("test_syslog_srv_ip {}".format(local_syslog_srv_ip))
-    
-    #Add Rsyslog destination for testing
+    logger.info("Configuring the DUT")
+    local_syslog_srv_ip = testbed["ptf_ip"]
+    logger.info("test_syslog_srv_ip %s", local_syslog_srv_ip)
+
+    # Add Rsyslog destination for testing
     duthost.shell("sudo config syslog add {}".format(local_syslog_srv_ip))
 
     yield
 
-    #remove the syslog configuration
+    # Remove the syslog configuration
     duthost.shell("sudo config syslog del {}".format(local_syslog_srv_ip))
 
+
 def test_syslog(duthost, ptfhost, config_dut, config_syslog_srv):
-    logger.info( "Starting syslog tests")
+    logger.info("Starting syslog tests")
     test_message = "Basic Test Message"
-    
-    #generate a syslog from the DUT
-    duthost.shell(" logger --priority INFO {}".format(test_message))
-
-    #Check syslog messages for the test message
-    result = ptfhost.shell("grep {} /var/log/syslog | grep \"{}\" | grep -v ansible".format(duthost.hostname, test_message))['stdout_lines']
-    pytest_assert(len(result) > 0, "Test syslog not seen on the server")
-
-
-    
-
 
-    
+    # Generate a syslog from the DUT
+    duthost.shell("logger --priority INFO {}".format(test_message))
 
+    # Check syslog messages for the test message
+    def _check_syslog():
+        result = ptfhost.shell("grep {} /var/log/syslog | grep \"{}\" | grep -v ansible"
+                               .format(duthost.hostname, test_message))["stdout_lines"]
+        return len(result) > 0
```

Comment:
could be replaced as` return test_message in result `

Reply:
Agree, this is more clear. Thanks!